src/main/navigation/navigation_fixedwing.c

```diff
@@ -463,17 +463,20 @@ int16_t applyFixedWingMinSpeedController(timeUs_t currentTimeUs)
 
 int16_t fixedWingPitchToThrottleCorrection(int16_t pitch)
 {
-    if (pitch > navConfig()->fw.pitch_to_throttle_thresh) {
-        // Positive pitch above threshold
-        return DECIDEGREES_TO_DEGREES(pitch - navConfig()->fw.pitch_to_throttle_thresh) * navConfig()->fw.pitch_to_throttle;
+    // Calculate base throttle correction from pitch moving average
+    const int16_t movingAverageCycles = 128; //Number of main loop cycles for average calculation
+    static int16_t averagePitch = (averagePitch * movingAverageCycles + pitch - averagePitch) / movingAverageCycles;
+    const int16_t baseThrottleCorrection = DECIDEGREES_TO_DEGREES(averagePitch) * navConfig()->fw.pitch_to_throttle;
+
+    // Calculate final throttle correction
```

Comment:
Not entirely sure what this block does. Is it to apply an extra throttle correction based on the difference between `pitch` and `averagePitch`? Say the airplane is pitched up 7 degrees (`pitch=-70`), with the average pitch being up 3 degrees (`averagePitch=-30`), then this subtracts 70 us from the throttle signal calculated above? Or take the case when `nav_fw_pitch2thr_threshold=0`, then this adds (`pitch` - `averagePitch`)*`fw.pitch_to_throttle` to the throttle value.

Can you explain the goal of this a bit?